akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -22,15 +24,18 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
   override def lookup = AddressUidExtension
 
   override def createExtension(system: ExtendedActorSystem): AddressUidExtension = new AddressUidExtension(system)
+
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
-  val longAddressUid: Long = {
-    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
-    //ThreadLocalRandom.current.nextLong()
-    ThreadLocalRandom.current.nextInt()
-  }
 
+  val longAddressUid: Long = ThreadLocalRandom.current.nextLong()
```

Comment:
You are right, I did like that at first, but then was tired in the end of the day and didn't remember the reasoning and thought "why do I need to handle them separately, the int will anyways just wraparound with the long value". I'll revert that.